app/routes.py

```diff
@@ -1,2 +1,145 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, request, jsonify
+from datetime import datetime
+import os, requests
+
+tasks_bp = Blueprint("task", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goal", __name__, url_prefix="/goals")
+
+#################### SLACK NOTIFICATIONS #################### 
+
+SLACK_TOKEN = os.environ.get('SLACK_TOKEN')
+CHANNEL_ID = os.environ.get('CHANNEL_ID')
+URL = 'https://slack.com/api/chat.postMessage'
+
+def send_notification_to_slack(msg):
+    headers = {"Authorization": f'Bearer {SLACK_TOKEN}'}
+    data = {"channel": CHANNEL_ID,"text": msg}
+    return requests.post(URL, headers=headers, data=data)
+
+######################## TASK ROUTES ######################## 
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+    request_body = request.get_json()
+    if len(request_body) < 3:
+        return jsonify(details = f'Invalid data'), 400
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])  
+    db.session.add(new_task)
+    db.session.commit()
+    send_notification_to_slack(f"New task created!")
+    return jsonify(task=new_task.to_json()), 201
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def view_all_tasks():
+    query_param_value = request.args.get("sort")
+    tasks = Task.sort(query_param_value) # sort method called on the class
+    view_tasks = [task.to_json() for task in tasks if tasks]
+    return jsonify(view_tasks)
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def view_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    return jsonify(task = task.to_json())
+
+@tasks_bp.route("/<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    updated_task = request.get_json()
+    if not updated_task:
+        return jsonify(None), 404
```

Comment:
Nice work accounting for invalid task data in the update action. You might consider returning a 400 rather than 404 since this is a Bad Request, rather than a resource that's not found.